### PR TITLE
Changes needed to support Calibra ticket (regarding SiteDataPlotter)

### PR DESCRIPTION
**Pull request by ZDu-IDM**
This PR addressed the following

Add utility tools to convert Windows file path to Linux file path for Slurm #1659
https://github.com/InstituteforDiseaseModeling/idmtools/issues/1659

SiteDataPlotter needs to be refactored to support SlurmPlatform #157
https://github.com/InstituteforDiseaseModeling/idmtools-calibra/issues/157

**Note**: this PR did all the required idmtools change for above calibra ticket.

Refer to PR in calibra
**_PR:_** Refactoring SiteDataPlotter to support LL for both COMPS and SLURM platform #165
https://github.com/InstituteforDiseaseModeling/idmtools-calibra/pull/165

### CODE REVIEW

**Review by devclinton — APPROVED** [review]

**Comment by devclinton** [comment]
@ZDu-IDM - Can you fix the linting issues?

**Comment by ZDu-IDM** [comment]
> @ZDu-IDM - Can you fix the linting issues?

Got all linting issues fixed, thanks!

**Comment by shchen-idmod** [comment]
This PR failed lot of tests: https://github.com/shchen-idmod/idmtools-1/actions/runs/3425970433/jobs/5707286888 with same reason. I suspect, we may change environment variable in host somehow with this PR.

File: idmtools_platform_comps/idmtools_platform_comps/comps_operations/experiment_operations.py [hunk 1]
```diff
@@ -467,3 +467,19 @@ def platform_list_asset(self, experiment: Experiment, **kwargs) -> List[Asset]:
...
         return assets
+
+    def create_sim_directory_map(self, experiment_id: Union[str, UUID]) -> Dict:
+        """
+        Build simulation working directory mapping.
+        Args:
+            experiment_id: experiment id
+
+        Returns:
+            Dict
```
Comment (shchen-idmod) [inline idmtools_platform_comps/idmtools_platform_comps/comps_operations/experiment_operations.py:478]: Can Dict comment more specific?

**Comment by ZDu-IDM** [comment]
> This PR failed lot of tests: https://github.com/shchen-idmod/idmtools-1/actions/runs/3425970433/jobs/5707286888 with same reason. I suspect, we may change environment variable in host somehow with this PR. Updated, tests are running on this branch as base and new test I added: [ZDu-IDM#1](https://github.com/ZDu-IDM/zdu_idmtools/pull/1). Failed tests seem appear after running my new testcases: test_create_sim_directory_map.py

[FIXED] make set/clear a pair usage, please verify, thanks!

File: idmtools_platform_comps/idmtools_platform_comps/comps_operations/experiment_operations.py [hunk 2]
```diff
@@ -467,3 +467,19 @@ def platform_list_asset(self, experiment: Experiment, **kwargs) -> List[Asset]:
...
+            experiment_id: experiment id
+
+        Returns:
+            Dict of simulation id as key and working dir as value
+        """
+        from idmtools_platform_comps.utils.linux_mounts import set_linux_mounts, clear_linux_mounts
+        set_linux_mounts(self.platform)
+        comps_exp = self.platform.get_item(experiment_id, ItemType.EXPERIMENT, raw=True)
+        comps_sims = comps_exp.get_simulations(QueryCriteria().select(['id', 'state']).select_children('hpc_jobs'))
+        clear_linux_mounts(self.platform)
```
Comment (shchen-idmod) [inline idmtools_platform_comps/idmtools_platform_comps/comps_operations/experiment_operations.py:484]: This line need after result dict.

**Review by shchen-idmod — CHANGES_REQUESTED** [review]

File: idmtools_platform_comps/idmtools_platform_comps/comps_operations/experiment_operations.py (hunk 2 above)
Comment (ZDu-IDM) [reply idmtools_platform_comps/idmtools_platform_comps/comps_operations/experiment_operations.py:485]: [Fixed] Good catch, thanks!

**Review by shchen-idmod — APPROVED** [review]
All tests passed. approved~

**Pull request merged**